### PR TITLE
[DOCS] How to validate data with an in memory checkpoint

### PR DESCRIPTION
**Pull request by Rachel-Reverie**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- Adds a how-to document "How to validate data with an in-memory Checkpoint."
- Adds sample scripts for the new how-to document.
- Places the sample scripts under test.
- Adds the new how-to document to the ToC
- Updates the how-to document for a process that this document replaces to direct readers to this document.

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [X] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [X] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [ ] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [ ] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/integration/docusaurus/validation/checkpoints/how_to_validate_data_with_a_python_configured_in_memory_checkpoint.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            "name": "update_data_docs",
+            "action": {"class_name": "UpdateDataDocsAction", "site_names": []},
+        },
+    ],
+    "validations": [
+        {
+            "batch_request": {
+                "datasource_name": "taxi_datasource",
+                "data_connector_name": "default_inferred_data_connector_name",
+                "data_asset_name": "yellow_tripdata_sample_2019-02.csv",
```
Comment (Shinnnyshinshin) [inline tests/integration/docusaurus/validation/checkpoints/how_to_validate_data_with_a_python_configured_in_memory_checkpoint.py]: I think this is a typo here: changing `data_asset_name` to `yellow_tripdata_sample_2019-02` should make tests pass

File: tests/integration/test_script_runner.py [hunk 2]
```diff
@@ -203,6 +203,18 @@ class IntegrationTestFixture:
...
         data_dir="tests/test_sets/taxi_yellow_tripdata_samples/first_3_files",
     ),
+    IntegrationTestFixture(
+        name="how_to_pass_an_in_memory_dataframe_to_a_checkpoint",
+        user_flow_script="tests/integration/docusaurus/validation/checkpoints/how_to_validate_data_with_a_yaml_configured_in_memory_checkpoint.py",
+        data_context_dir="tests/integration/fixtures/no_datasources/great_expectations",
+        data_dir="tests/test_sets/taxi_yellow_tripdata_samples/first_3_files",
+    ),
+    IntegrationTestFixture(
+        name="how_to_pass_an_in_memory_dataframe_to_a_checkpoint",
```
Comment (Shinnnyshinshin) [inline tests/integration/test_script_runner.py]: the `name` field here is used to distinguish `IntegrationTestFixtures` from one another. So I'd request these to be changed to something like `how_to_pass_an_in_memory_dataframe_to_a_checkpoint_yaml_config` and `how_to_pass_an_in_memory_dataframe_to_a_checkpoint_python_config`

Comment (Rachel-Reverie) [reply tests/integration/test_script_runner.py]: Thank you, updating that now.

File: tests/integration/docusaurus/validation/checkpoints/how_to_validate_data_with_a_python_configured_in_memory_checkpoint.py (hunk 1 above)
Comment (Rachel-Reverie) [reply tests/integration/docusaurus/validation/checkpoints/how_to_validate_data_with_a_python_configured_in_memory_checkpoint.py]: I've updated it to use the same name that was used in the YAML version of the config.

File: tests/integration/docusaurus/validation/checkpoints/how_to_validate_data_with_a_yaml_configured_in_memory_checkpoint.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+)
+from great_expectations.core.run_identifier import RunIdentifier
+from great_expectations.data_context.types.base import CheckpointConfig
+from great_expectations.data_context.types.resource_identifiers import (
+    ValidationResultIdentifier,
+)
+
+# Import and setup for working with YAML strings:
+# <snippet>
+from ruamel import yaml
```
Comment (Shinnnyshinshin) [inline tests/integration/docusaurus/validation/checkpoints/how_to_validate_data_with_a_yaml_configured_in_memory_checkpoint.py]: I also saw that the `lint` stage is failing, and it looks the issue is having this import too far down. 

can we pair to see if moving it toward the top of the script would be an option? I can help you get set up with the linter locally as well.

Comment (Rachel-Reverie) [reply tests/integration/docusaurus/validation/checkpoints/how_to_validate_data_with_a_yaml_configured_in_memory_checkpoint.py]: I would appreciate that.  Feel free to set a meeting when convenient for your schedule.

File: docs/guides/validation/advanced/how_to_validate_data_with_an_in_memory_checkpoint.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+### 6. Check your Data Docs
+
+Once you have run your script you can verify that it has worked by checking your Data Docs for new results.
+
+## Notes
+
+To view the full example scripts used in this documentation, see:
+- [how_to_validate_data_with_a_yaml_configured_in_memory_checkpoint.py]
+- [how_to_validate_data_with_a_python_configured_in_memory_checkpoint.py]
```
Comment (Shinnnyshinshin) [inline docs/guides/validation/advanced/how_to_validate_data_with_an_in_memory_checkpoint.md]: One last point : I think these links would need to be updated

**Review by Shinnnyshinshin — APPROVED** [review]
thank you @Rachel-Reverie looks good to me other than the links to the scripts at the end of the doc

**Review by donaldheppner — APPROVED** [review]

**Comment by Shinnnyshinshin** [comment]
Hi @Rachel-Reverie I accidentally merged this branch with my working branch, so now I have a really big diff with `/develop`. Doing my part to ensure this gets merged in :)

**Pull request merged**